vm/2.0/modules/ROOT/pages/vm-propagate-messages.adoc

```diff
@@ -0,0 +1,188 @@
+= VM Connector Propagate Message Attributes Example - Mule 4
+
+The following example illustrates how you can propagate message attributes through a VM queue via a message payload. The first flow takes incoming HTTP message attributes and additionally creates a custom attribute of its own, and then adds both of these attributes into a message payload so the *VM Publish* operation can publish the message to a VM queue. In the second flow, the *VM Listener* source receives the message payload from the previous flow and sets its own message attributes. Subsequently, the message payload is finally stored in an object store.
+
+The following screenshot shows the Anypoint Studio app flow for this example:
+
+.VM Connector Propagate message attributes flow
+image::vm-example-messageflow.png[VM Connector example flow]
+
+== Create the VM Publish Flow
+
+The VM Publish flow first listens for incoming HTTP message attributes, then transforms the message attributes into a JSON structure message, and finally the *VM Publish* operation publishes the message in a VM queue. +
```

Comment:
Publish or publish?
Is this a formal name? I'm guessing not....
GLOBAL; I will stop marking.

Reply:
This sentence is unbalanced, because the flow is the subject for the first two actions and the operation for the third. So I think we need one of the following alternatives:
```suggestion
The VM publishing flow first listens for incoming HTTP message attributes, then transforms the message attributes into a JSON structure message, and finally publishes the message in a VM queue. +
```
OR
```suggestion
The VM publishing flow first listens for incoming HTTP message attributes and then transforms the message attributes into a JSON structure message. Finally, the *VM Publish* operation publishes the message in a VM queue. +
```
OR remove "first," "then," and "finally" from either version.